macros/as_primitive.sql

```diff
@@ -1,26 +1,38 @@
-{%- macro as_primitive(t, c) -%}
+{%- macro as_primitive(t, c, try_casting=False) -%}
     {% set query = "
-        select 
-            typeof(" ~ c ~ ") as type, 
-            ifnull(len(split(" ~ c ~ ", '.')[1]), 2) as precision 
+        select
+            ifnull(try_parse_json(" ~ c ~ "), " ~ c ~ ") as parsed,
+            typeof(parsed) as type, 
+            ifnull(len(split(parsed, '.')[1]), 2) as scale 
         from " ~ t ~ " 
-        where type is not null 
+        where ifnull(type, 'NULL_VALUE') != 'NULL_VALUE'
```

Comment:
is the point here that we can get both the SQL null and the JSON null?

Reply:
Good question. I believe the answer is yes. Basically, you can get a scenario where the key `c` exists in the JSON, but is value is null and in that case you get a type of 'NULL_VALUE'. In the other scenario the key `c` doesn't exists in the JSON and in that case the type = `null`

Reply:
let's make a comment about that, since it's not obvious, then merge?

Reply:
Will do